Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Mesh/ModelReloaderSystemInterface.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/Event.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/RTTI/RTTI.h>
+
+namespace AZ
+{
+    namespace Data
+    {
+        template<class T>
+        class Asset;
+    }
+
+    namespace RPI
+    {
+        class ModelAsset;
+    }
+
+    namespace Render
+    {
+        using ModelReloadedEvent = Event<const Data::Asset<RPI::ModelAsset>>;
+
+        //! A system that handles reloading the hierarchy of model assets in the correct order
+		class ModelReloaderSystemInterface
+        {
+        public:
+            AZ_RTTI(AZ::Render::ModelReloaderSystemInterface, "{E7E05B1F-8928-4A1B-B75D-3D5433E65BCA}");
+
+            ModelReloaderSystemInterface()
+            {
+                Interface<ModelReloaderSystemInterface>::Register(this);
+            }
+
+            virtual ~ModelReloaderSystemInterface()
+            {
+                Interface<ModelReloaderSystemInterface>::Unregister(this);
+            }
+
+            static ModelReloaderSystemInterface* Get()
+            {
+                return Interface<ModelReloaderSystemInterface>::Get();
+            }
```

Comment:
This is the interface class, it's not suppose to register itself, but only provide the interface.

Some implementation of this interface has to register itself as the implementer of this interface, in this case `class ModelReloaderSystem`, and there is a easy helper to do this:

````
        class ModelReloaderSystem
            : public AZ::Interface<ModelReloaderSystemInterface>::Registrar
````
This will automatically register `ModelReloaderSystem` as the implementor of `ModelReloaderSystemInterface` so everybody doing `AZ::Interface<ModelReloaderSystemInterface>::Get()` will work the moment `ModelReloaderSystem` is instanced.

Reply:
I will try that, thanks!